src/inference/dream/estimator.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var util = require('../../util');
+var paramStruct = require('../../params/struct');
+
+// This estimator currently makes the following assumptions:
+
+// 1. The model contains exactly one `mapData`.
+
+// 2. Either a) each element of the data array is observed (passed to
+// `observe` as its second argument) exactly once in the corresponding
+// observation function, or b) each element of the data array is
+// itself an array, of which each element is observed exactly once,
+// and in the order in which they appear in the array. See below for
+// examples of each of these.
```

Comment:
I don't have a clear idea about how best to implement this yet, so I'd be in favour of tackling it later. Of course, someone else may already have a good approach, but even so, doing it as a separate PR seems like a good way to go to me. (Unless it's super easy.)